vars/warnAboutDeprecatedChartConfig.groovy

```diff
@@ -19,15 +21,17 @@ def call(Map<String, String> params) {
     sh 'rm -f check-helm-api-version.sh'
   }
 
-  try {
-    sh """
-    chmod +x check-deprecated-charts.sh
-    ./check-deprecated-charts.sh $product $component
-    """
-  } catch(ignored) {
-    WarningCollector.addPipelineWarning("deprecated_helmcharts", "Please upgrade base helm charts to latest. See releases on the chart repo for latest updates, example: https://github.com/hmcts/chart-java/releases", LocalDate.of(2023, 03, 29))
-  } finally {
-    sh 'rm -f check-deprecated-charts.sh'
+  sh 'chmod +x check-deprecated-charts.sh'
+
+  helmChartDeprecationConfig.each { chart, deprecatedVersions ->
+    deprecatedVersions.each { deprecation ->
+      try {
+        sh """./check-deprecated-charts.sh $product $component $chart $deprecation.version """
+      } catch (ignored) {
+        WarningCollector.addPipelineWarning("deprecated_helmcharts", "Version of $chart helm chart below $deprecation.version is deprecated, please upgrade to latest release https://github.com/hmcts/chart-$chart/releases", LocalDate.parse(deprecation.date_deadline))
```

Comment:
this just assumes an error means the chart is deprecated which is probably fine, but just calling out